client/rpc_client.go

```diff
@@ -144,10 +150,14 @@ func ClientFromConfig(c *Config) (*RPCClient, error) {
 		dispatch:   make(map[uint64]seqHandler),
 		shutdownCh: make(chan struct{}),
 	}
-	client.dec = codec.NewDecoder(client.reader,
-		&codec.MsgpackHandle{RawToString: true, WriteExt: true})
-	client.enc = codec.NewEncoder(client.writer,
-		&codec.MsgpackHandle{RawToString: true, WriteExt: true})
+	handle := &codec.MsgpackHandle{
+		WriteExt: true,
+		BasicHandle: codec.BasicHandle{
+			TimeNotBuiltin: !c.MsgpackUseNewTimeFormat,
+		},
+	}
+	client.dec = codec.NewDecoder(client.reader, handle)
+	client.enc = codec.NewEncoder(client.writer, handle)
```

Comment:
I think it's safer to create 2 separate `handle` for the encoder and the decoder. While I don't think it's an issue now as we don't keep any state in the handle, but I think it introduces a possible foot gun if we do so in the future.